glotaran/project/dataclasses.py

```diff
@@ -0,0 +1,125 @@
+"""Contains helper methods for dataclasses."""
+from __future__ import annotations
+
+import dataclasses
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import Callable
+    from typing import TypeVar
+
+    DefaultType = TypeVar("DefaultType")
+
+
+def exclude_from_dict_field(
+    default: DefaultType = dataclasses.MISSING,  # type:ignore[assignment]
+) -> DefaultType:
+    """Create a dataclass field with which will be excluded from ``asdict``.
+
+    Parameters
+    ----------
+    default : DefaultType
+        The default value of the field.
+
+    Returns
+    -------
+    DefaultType
+        The created field.
+    """
+    return dataclasses.field(default=default, metadata={"exclude_from_dict": True})
+
+
+def file_representation_field(
+    target: str,
+    loader: Callable[[str], Any],
+    default: DefaultType = dataclasses.MISSING,  # type:ignore[assignment]
+) -> DefaultType:
+    """Create a dataclass field with target and loader as metadata.
+
+    Parameters
+    ----------
+    target : str
+        The name of the represented field.
+    loader : Callable[[str], Any]
+        A function to load the target field from a file.
+    default : DefaultType
+        The default value of the field.
+
+    Returns
+    -------
+    DefaultType
+        The created field.
+    """
+    return dataclasses.field(default=default, metadata={"target": target, "loader": loader})
+
+
+def asdict(dataclass: object) -> dict[str, Any]:
```

Comment:
Would prefer to avoid `dataclass` as a variable name. Also the function name could be more specific. 
Assuming the module is rename dataclass_helpers this could be rnamed dataclass_to_dict or dataclass_as_dict and perhaps even make them semi private by preceeding with a `_`.